jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1272,7 +1273,11 @@ public void succeeded()
                 httpChannel.lockedStreamSendCompleted(true);
             }
             if (callback != null)
-                httpChannel._serializedInvoker.run(callback::succeeded);
+            {
+                // We cannot use the SerializedInvoker here because this may be called from a blocking call within
+                // the application which is running inside the SerializedInvoker, then it is a deadlock.
+                callback.succeeded();
```

Comment:
This was being called from within the `jakarta.servlet.ReadListener.onAllDataRead()` doing a blocking `ServletOutputStream.close()`.

Because `ReadListener.onAllDataRead` is invoked from inside the `_serializedInvoker` we would never be able to succeed the callback if we use `_serializedInvoker` here as well.